example/set.wren

```diff
@@ -0,0 +1,110 @@
+class Set {
+  new {
+    _list = []
+    _clean = true
+  }
+
+  new (list) {
+    if (list is List) {
+      _list = list
+      _clean = false
+      cleanup
+    } // raise error?
+  }
+
+  cleanup {
+    // Removes duplicates in the underlying list.
+    if (!_clean) {
+      var newList = []
+      for (element in _list) {
+        if (!newList.contains(element)) newList.add(element)
+      }
+      _list = newList
+      _clean = true
+    }
+  }
+
+  add(element) {
+    _clean = false
+    _list.add(element)
+  }
+
+  remove(element) {
+    cleanup // Remove duplicates, so we can return early upon deletion.
+    for (i in 0.._list.count) {
+      if (_list[i] == element) {
+        _list.removeAt(i)
+        return
+      }
+    }
+  }
+
+  contains(element) {
+    return _list.contains(element)
+  }
+
+  count {
+    cleanup
+    return _list.count
+  }
+
+  iterate(i) {
+    cleanup
+    if (i == null) {
+      if (count > 0) return 0
+      return null
+    }
+    if (i < count || i >= count) return false
+    return i + 1
+  }
+
+  iteratorValue(i) {
+    cleanup
+    return _list[i]
+  }
+
+  map(f) {
+    return new Set(_list.map(f))
+  }
+
+  where(f) {
+    return new Set(_list.where(f))
+  }
+
+  |(that) {
+    // Union
```

Comment:
Yeah, I did [something similar](http://magpie-lang.org/program-structure.html#doc-comments) for my last language.

I think I had convinced myself that other core lib methods were documented on the outside, but now I see that they aren't documented at all (well, except for on the site).

If we get to the point where we start hacking on an automated doc generator, we can worry more seriously about it then.